src/helpers/constants.h

```diff
@@ -18,7 +19,7 @@
     }\
 
 #define REGISTER_CONSTANT(name)\
-    register_property<Fmod, int>(#name,&Fmod::set_##name, &Fmod::get_##name, name);\
+    BIND_CONSTANT(name)\
```

Comment:
`BIND_CONSTANT` is a macro defined [here](https://github.com/godotengine/godot/blob/6236a688b77b11de3570c94f77069049d65d25fd/core/object/class_db.h#L375) that calls `ClassDB::bind_integer_constant`, defined [here](https://github.com/godotengine/godot/blob/6236a688b77b11de3570c94f77069049d65d25fd/core/object/class_db.cpp#L546). It basically grabs property name, the class it belongs to, and its value, and then registers it all to the ClassDB. It must be an integer constant.

Registering a property, which the old code was doing, is fundamentally a different thing since it *technically* defines a potentially mutable value for which memory is allocated with every new instance of the object. So I would double check whether making this change actually causes any potential behavioral problems.

In the case that you *do* need a property registration, I suspect GDExtensions should offer an `ADD_PROPERTY()` macro since it's designed to be a mostly drop-in replacement for the Godot module API and that's what they do (peek into any classes `_bind_methods()` function in Godot's source code and you'll find numerous examples of how it should be used).

Reply:
Thanks a lot! I tried `ADD_PROPERTY` before but I do not get it to work. The old implementation passed in a pointer to the setter and getter, while the new implementation of `ADD_PROPERTY` expects a string name of the setters/getters. Will revisit this later., unless somebody else has a better idea.

Reply:
Well in that case, they actually need to be constant so it's fine. 
At the time I had those  integers registered as properties because I couldn't find a way to register constants. So I just register a fake getter/setter so the value can't be modified.

Reply:
@CedNaru thank you for clarifying this!